Archive/Imo/Imo2008Q2.lean

```diff
@@ -106,12 +107,7 @@ theorem imo2008_q2b : Set.Infinite rationalSolutions := by
             exact ⟨rfl, rfl, rfl⟩
         · have hg : -z = g (x, y, z) := rfl
           rw [hg, hz_def]; ring
-      have h₂ : q < t * (t + 1) := by
-        calc
-          q < q + 1 := by linarith
-          _ ≤ t := le_max_left (q + 1) 1
-          _ ≤ t + t ^ 2 := by linarith [sq_nonneg t]
-          _ = t * (t + 1) := by ring
+      have h₂ : q < t * (t + 1) := by linear_combination sq_nonneg t + le_max_left (q + 1) 1
```

Comment:
This is a clear improvement, imho.

Reply:
This golf works equally well without changing `linarith` to `linear_combination`; moved it to #19319.